DFC.ServiceTaxonomy.GraphSync/GraphSyncers/ValidateAndRepairGraph.cs

```diff
@@ -195,22 +192,17 @@ public async Task<(bool validated, string failureReason)> ValidateContentItem(Co
 
             foreach (ContentTypePartDefinition contentTypePartDefinition in contentTypeDefinition.Parts)
             {
-                string partTypeName = contentTypePartDefinition.PartDefinition.Name;
-                string partName = contentTypePartDefinition.Name;
-                if (!_partSyncers.TryGetValue(partTypeName, out var partSyncer)
-                    && (partName == contentTypePartDefinition.ContentTypeDefinition.Name ||
-                        contentTypePartDefinition.PartDefinition.Fields.Any(f => GroupingFields.Contains(f.FieldDefinition.Name))))
-                {
-                    partSyncer = _partSyncers["Eponymous"];
-                }
+                IContentPartGraphSyncer partSyncer = _partSyncers.SingleOrDefault(ps =>
+                    ps.CanHandle(contentTypePartDefinition.ContentTypeDefinition.Name,
+                        contentTypePartDefinition.PartDefinition));
 
                 if (partSyncer == null)
                 {
                     // part doesn't have a registered IContentPartGraphSyncer, so we ignore it
```

Comment:
Could log out ContentTypeDefinition.Name here for diagnostics in case we have something setup incorrectly?

Reply:
Good idea, added.